lang/java/reef-runtime-local/src/main/java/org/apache/reef/runtime/local/client/LocalJobSubmissionHandler.java

```diff
@@ -108,10 +110,22 @@ public void onNext(final JobSubmissionEvent t) {
         this.configurationSerializer.toFile(driverConfiguration,
             new File(driverFolder, this.fileNames.getDriverConfigurationPath()));
         this.driverLauncher.launch(driverFolder);
+        this.applicationId = t.getIdentifier();
```

Comment:
in `HDInsightJobSubmissionHandler`, variable `appId` is a shortcut for the chain call `this.hdInsightInstance.getApplicationID().getApplicationId()` that we would otherwise invoke multiple times; still, we do not want to assign that value to `this.applicationId` until we are sure we've successfully submitted the application. There is no need for such variable in Mesos and other handlers.